avocado/core/dispatcher.py

```diff
@@ -14,6 +14,7 @@
 
 """Extensions/plugins dispatchers."""
 
+import logging
```

Comment:
Well the `stdout_claim_by` is ugly (but less ugly than before) method. The way I wanted to solve this in the logging refactor was to split the logging streams and replace the loggers based on the input.

In normal execution it'd enable `avocado.app.ui`. On `--json` it'd disable `avocado.app.ui` and instead use `avocado.app.json` to stdout. The same way `avocado.app.xunit` would try to replace the default UI. The switching would be done by `avocado.core.output` so it could remember if it was already switched and by whom, but it only stayed as "TODO" item. Feel free to incorporate it, though :wink: